### ISSUE TITLE
Clicking Debug Freezes Up 

### ISSUE THREAD

**Issue by asebak** [asebak/raspberrypi-intellij#23]
The debugger freezes up the UI until it finishes deploying.

**Comment by asebak** [comment]
@araad you want to take a look at this issue? I can show you exactly where this code is.

**Comment by araad** [comment]
Yeah, I can take a look at it tonight

**Comment by asebak** [comment]
it's in this file
where debug and run happens
 https://github.com/asebak/raspberrypi-intellij/blob/master/src/com/atsebak/raspberrypi/commandline/AppCommandLineState.java

**Comment by asebak** [comment]
@araad 
Exact line of code is

```
if (isDebugMode) {
                final String initializeMsg = String.format(DEBUG_TCP_MESSAGE, configuration.getRunnerParameters().getPort());
                //this should wait until the deployment states that it's listening to the port
                while (!outputForwarder.toString().contains(initializeMsg)) {
                }
                closeOldSessionAndDebug(project, configuration.getRunnerParameters());
            }
```

The while loop needs to waiting for the debug message to show before attaching the debugger.

**Comment by araad** [comment]
I created a pull request with a partial fix that wont freeze up the UI, the other part of the task is to fix exceptions related to not synchronizing reads between threads. Can you check it out?

**Comment by araad** [comment]
The plugin will still run, just that the user will see exceptions in the event log

**Comment by asebak** [comment]
fixed by #24
